docs/reverse.rst

```diff
@@ -111,3 +132,164 @@ in mailu.env:
   WEBROOT_REDIRECT=/sso/login
 
 Using the above configuration, Traefik will proxy all the traffic related to Mailu's FQDNs without requiring duplicate certificates.
+
+
+Traefik as reverse proxy (different host)
+-----------------------------------------
+
+This example is for when Traefik (reverse proxy) runs on a different server than the Mailu server.
+This example makes use of the File configuration method (File Provider) of Traefik.
+It makes use of a single static configuration file and one or more dynamic configuration files.
+
+This example uses V3 of Traefik. V2 and V3 of Traefik have some differences in the configuration method.
+
+The contents for the static configuration file. The static configuration file must be provided as argument to Traefik.
+
+.. code-block:: yaml
+
+  #STATIC CONFIGURATION FILE
+  #Below value for 'directory' is the location where the dynamic configuration files reside:
+  #When a change is made in this folder, Traefik automatically loads or reloads it.
+  providers:
+    file:
+      directory: "/etc/traefik/conf"
+
+  entryPoints:
+    web:
+      address: :80
+    websecure:
+      address: :443
+    mailu-smtp:
+      address: :25
+    mailu-imaps:
+      address: :993
+    mailu-pop3s:
+      address: :995
+    mailu-smtps:
+      address: :465
+    mailu-sieve:
+      address: :4190
+
+  #Optional, enables access logging at:
+  accessLog:
+    filePath: "/var/log/traefik_access.log"
+
+  #Optional, enables normal logging at:
+  log:
+    level: INFO
+    filePath: "/var/log/traefik.log"
+
+
+This is the contents for the dynamic configuration. You can use and filename you want.
+The extension must end with .yml and the file must be placed in the configured directory for
+dynamic configuration files.
+
+
+.. code-block:: yaml
+
+  http:
+    routers:
+      mailu-web:
+        entryPoints:
+          - web
+        rule: "Host(`mail.example.com) || PathPrefix(`/.well-known/acme-challenge/`))"
```

Comment:
This is too broad if we don't want to break it for other vhosts... is there a fancy way to declare all a variable containing all the hostnames we want to match so that we could reuse the same in `mailu-websecure`?

Reply:
You also used this?
https://github.com/Mailu/Mailu/blob/341afeb57d266c1d990d1c41a6c57f0d2852117a/docs/reverse.rst?plain=1#L59

With Traefik V3 you cannot define multiple hosts in Host in the rule. All matchers take a single value:
https://doc.traefik.io/traefik/v3.0/migration/v2-to-v3-details/#new-v3-syntax-notable-changes

So you would simply add a lot of `Host('yet another hostname from HOSTNAMES')` to your rule.

I don't see a problem with this. For all other vhosts Traefik can use the TLS challenge (TLS-ALPN-01) which uses port 443. So this will not lead to any issues. It's as easy as configuring:

```
certificatesResolvers:
  myresolver:
    acme:
      email: redacted@redacted.redacted
      storage: /etc/traefik/acme/acme.json
      caServer: "https://acme-v02.api.letsencrypt.org/directory"
      #caServer: "https://acme-staging-v02.api.letsencrypt.org/directory"
      tlsChallenge: {}
#      httpChallenge:
#        # used during the challenge
#        entryPoint: web
```
Below in comment you see the HTTP challenge which cannot be used because Mailu uses it. With above config, Traefik uses TLS challenge via port 443 and Mailu uses port 80 for the HTTP challenge.

I've this running on my homelab.

Reply:
OK, let's add a comment to spell out that we need the rule to let the requests through for all hostnames Mailu will use (the OR is not an accident)